### PR TITLE
[Filebeat] Check content type when reading s3 files

### PR DESCRIPTION
**Pull request by kaiyan-sheng**
When file name has `.gz` suffix but with text/plain content type, `newS3BucketReader` function will fail when using s3 input in Filebeat. Instead of simply checking file name, check the actual content type from the response and then decide how to build the new reader. 

## How to test it:
Upload file to an S3 bucket and change the file metadata property to test this PR:
Screen Shot 2020-01-07 at 7 48 59 AM

Upload a test1.txt.gz file and change content type to `text/plain`, s3 input should still be able to read the file.

closes https://github.com/elastic/beats/issues/15225

### CODE REVIEW

**Comment by kaiyan-sheng** [comment]
jenkins, test this please

**Comment by ynirk** [comment]
For information, AWS Cloudtrail set the content type to `application/json` and i had the same problem. 
Applying this patch to filebeat 7.5.1 solved the issue

File: x-pack/filebeat/input/s3/input.go [hunk 1]
```diff
@@ -440,16 +440,30 @@ func (p *s3Input) newS3BucketReader(svc s3iface.ClientAPI, s3Info s3Info) (*bufi
 		return nil, errors.New("s3 get object response body is empty")
 	}
 
+	// Check content-type
+	if resp.ContentType != nil {
+		switch *resp.ContentType {
+		case "text/plain", "text/plain; charset=utf-8":
+			return bufio.NewReader(resp.Body), nil
```
Comment (mtojek) [inline x-pack/filebeat/input/s3/input.go]: Isn't it the same as default case?

Comment (kaiyan-sheng) [reply x-pack/filebeat/input/s3/input.go]: yeah, good catch. I will remove this case section! Thanks!

**Comment by kaiyan-sheng** [comment]
jenkins, test this please

**Comment by kaiyan-sheng** [comment]
ci failures are not related.

**Comment by kaiyan-sheng** [comment]
This PR will be tested when testing https://github.com/elastic/beats/pull/15370

**Pull request merged**